mesonbuild/msubprojects.py

```diff
@@ -65,23 +96,22 @@ def done(self, wrap_name: str, log_queue: T.List[T.Tuple[mlog.TV_LoggableList, T
 
 
 class Runner:
-    def __init__(self, logger: Logger, r: Resolver, wrap: PackageDefinition, repo_dir: str, options: argparse.Namespace) -> None:
+    def __init__(self, logger: Logger, r: Resolver, wrap: PackageDefinition, repo_dir: str, options: 'Arguments') -> None:
         # FIXME: Do a copy because Resolver.resolve() is stateful method that
         # cannot be called from multiple threads.
         self.wrap_resolver = copy.copy(r)
-        self.wrap_resolver.current_subproject = 'meson'
         self.wrap_resolver.dirname = os.path.join(r.subdir_root, wrap.directory)
         self.wrap = self.wrap_resolver.wrap = wrap
         self.repo_dir = repo_dir
         self.options = options
-        self.run_method = options.subprojects_func.__get__(self)
-        self.log_queue = []
+        self.run_method: T.Callable[[], bool] = options.subprojects_func.__get__(self) # type: ignore
```

Comment:
Wait, what? why are we assigning the `descriptor get to this? I know you don't write this, but WAT.

Reply:
Yea, I don't quite understand it or how it works.